lib/sqlalchemy/util/_concurrency_py3k.py

```diff
@@ -14,9 +20,26 @@
     # Issue for context: https://github.com/python-greenlet/greenlet/issues/173
 
     class _AsyncIoGreenlet(greenlet.greenlet):
-        def __init__(self, fn, driver):
-            greenlet.greenlet.__init__(self, fn, driver)
-            self.driver = driver
+        if copy_context is None or greenlet.__version__ < "0.4.17":
+
+            def __init__(self, fn, driver):
+                greenlet.greenlet.__init__(self, fn, driver)
+                self.driver = driver
+
+        elif greenlet.__version__ == "0.4.17":
```

Comment:
can you fill me in on what is specific about 0.4.17 here?   I would think that we would disregard using new behavior for this specific version.   also version comparison would need to be done more robustly, however I would hope that we can use a simple duck-typing approach:

```
    if hasattr(self, "gr_context"):
        self.gr_context = copy_context()

```

Reply:
Oh sure - greenlet 0.4.17 included python-greenlet/greenlet#172, which starts new greenlets with empty contexts. #5615 is only caused by greenlet 0.4.17.

I think the duck-typing approach would work fine. I'll update this PR as a draft proposal.

Reply:
Personally I think it would be fine to constraint the install to `gr_context != 0.4.17`